v-next/hardhat/src/internal/builtin-plugins/artifacts/artifacts-manager.ts

```diff
@@ -6,15 +6,176 @@ import type {
   CompilerOutput,
 } from "../../../types/artifacts.js";
 
-import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import path from "node:path";
+
+import {
+  assertHardhatInvariant,
+  HardhatError,
+} from "@ignored/hardhat-vnext-errors";
+import {
+  exists,
+  getAllFilesMatching,
+  getFileTrueCase,
+  readJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+
+export const BUILD_INFO_DIR_NAME = "build-info";
 
 export class ArtifactsManagerImplementation implements ArtifactsManager {
```

Comment:
There are a large nubmer of changes in this file.
My procedure here was to run the Solidity test task until it hit a not implemented exception and then keep pulling across the v2 Artifacts versions of methods into this file.
Where I could I used our utils.
The changes are encapsulated in this file (which can be deleted as soon as we have a fuller artifacts manager).

Reply:
I have added a comment that this is a shim for development and will be deleted with the new build system.
I don't think displaying warning messages justifies the effort.